ttnn/cpp/ttnn/operations/data_movement/transpose/device/transpose_op.cpp

```diff
@@ -147,9 +153,15 @@ std::vector<ttnn::TensorSpec> Transpose::compute_output_specs(const std::vector<
         if (this->dim == TransposeOpDim::WH) {
             const auto& input_padded_shape = input_tensor.get_padded_shape();
             ShardSpec shard_spec = input_tensor.shard_spec().value();
-            shard_spec.shape[0] = shard_spec.shape[0] / input_padded_shape[-2] * input_padded_shape[-1];
```

Comment:
a note to myself. a user must be mindful if transposed shard will fit the memory :)